gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -341,6 +341,15 @@ def self.run_sig(hook_mod, method_name, original_method, declaration_block)
         Signature.new_untyped(method: original_method)
       end
 
+    current_method = signature.method.owner.instance_method(method_name)
+    file_path, line = current_method.source_location
+
+    if !@was_ever_final_names[method_name] && original_method != current_method &&
```

Comment:
I'd argue that this logic should be encapsulated inside the `unwrap_method` call, since that would make sure regardless of how `unwrap_method` is called, this is always checked.